### PR TITLE
Widen the range of allowed cell broadcast topics

### PR DESCRIPTION
**Pull request by Flohack74**
In fact cell broadcast can subscribe to topics up to 9999 so lets remove that restriction

### CODE REVIEW

File: ofono/src/smsutil.c [hunk 1]
```diff
@@ -4565,7 +4566,7 @@ GSList *cbs_optimize_ranges(GSList *ranges)
 
 	range = NULL;
 
-	for (i = 0; i <= 999; i++) {
+	for (i = 0; i <= CBS_MAX_TOPIC; i++) {
```
Comment (monich) [inline ofono/src/smsutil.c:4569]: Watch for bitmap array overflow (125 = 999/8+1)

File: ofono/src/cbs.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ *  it under the terms of the GNU General Public License version 2 as
+ *  published by the Free Software Foundation.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ */
+
+#define CBS_MAX_TOPIC 9999
```
Comment (lduboeuf) [inline ofono/src/cbs.h]: shouldn't that be part of "include/cbs.h" ?

**Review by lduboeuf — CHANGES_REQUESTED** [review]

File: ofono/src/cbs.h (hunk 2 above)
Comment (Flohack74) [reply ofono/src/cbs.h]: No, they have the header files in parallel, I am just mimicking what I see ;)

**Comment by lduboeuf** [comment]
Tested on Xenial with success. We can set/unset Topics 0->9999
Maybe not related at all and intentional, but we can't set the Powered property to false.

**Comment by lduboeuf** [comment]
We will also need to add CMAS PRESIDENTIAL ALERT as default subscribed channel:
see https://github.com/ubports/ofono/pull/34/commits/fae493c40c502c39db826e881359efc379864978

**Comment by lduboeuf** [comment]
Note that i've just tested that PR, but i have issues for whatever reason ( phonesim app crash )

By using https://github.com/ubports/ofono/pull/40 which is just a cherry-pick from Mer upstream, it works....

**Comment by gber** [comment]
Obsolete, this might be addressed by upgrading to a newer upstream release.

**Pull request closed without merging**